tools/buildgen/extract_metadata_from_bazel_xml.py

```diff
@@ -410,8 +437,8 @@ def _extract_cc_tests(bazel_rules):
     return list(sorted(result))
 
 
-def _filter_cc_tests(tests):
-    """Filters out tests that we don't want or we cannot build them reasonably"""
+def _exclude_unwanted_cc_tests(tests):
+    """Filters out bazel tests that we don't want to run with other build systems or we cannot build them reasonably"""
 
     # most qps tests are autogenerated, we are fine without them
```

Comment:
It would be nice to move the specific cases we're excluding here out of this file and into a data file, so that the specific inputs can be modified without changing this script.

Alternatively, why do we still need any of the C/C++ tests to run in non-bazel builds?  Couldn't we just ignore all tests here?

Reply:
In the long run, we probably can remove some C/C++ tests in  non-bazel builds but there are some concerns:

- if the Make/cmake builds don't have any tests at all, how can we be sure what they've build actually works? (so we need at least some "smoke tests"). Btw, the build flags between bazel / non-bazel flag might differ (sometimes it's for interesting reasons that are not easy to overcome) which makes it even more important to have at least some tests for make/cmake.

- our current portability suite actually runs make/cmake tests and we'd need to migrate it to use bazel first (which brings the question whether it's possible to get a good enough coverage of different compilers / distros etc, with bazel - it's something we haven't really tried yet).

- some things that are considered "tests" in build.yaml based builds are actually binaries you'd want to be able to build anyway (qps_json_worker, interop_client, interop_server, grpc_cli) so it's unclear how much make/cmake simplification we would gain by removing just some (but not all) tests.

Reply:
Please document these concerns in TODO comments.

Reply:
Documented the reasons/concerts in a comment at "Step 2" (extracting tests).